src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -432,16 +446,17 @@ uct_ib_mlx5_devx_reg_ksm_data(uct_ib_mlx5_md_t *md,
 {
     uct_ib_mlx5_devx_mr_t *mr = &memh->mrs[mr_type];
     void *address             = uct_ib_mlx5_devx_memh_base_address(memh);
+    uint64_t iova             = uct_ib_mlx5_devx_memh_iova(memh, iova_offset);
 
     if (memh->super.flags & UCT_IB_MEM_MULTITHREADED) {
         return uct_ib_mlx5_devx_reg_ksm_data_mt(md, address,
                                                 (uint64_t)address + iova_offset,
                                                 atomic, mkey_index, reason,
                                                 mr->ksm_data, mr_p, mkey);
     } else {
-        return uct_ib_mlx5_devx_reg_ksm_data_contig(md, address, iova_offset,
-                                                    atomic, mkey_index, reason,
-                                                    mr, mr_p, mkey);
+        return uct_ib_mlx5_devx_reg_ksm_data_contig(md, address, iova, atomic,
```

Comment:
shouldn't we pass iova to uct_ib_mlx5_devx_reg_ksm_data_mt (line 452) as well? instead of `(uint64_t)address + iova_offset` ?

Reply:
I think not, because data_contig is aligning down the base address while it was not the logic in `uct_ib_mlx5_devx_reg_ksm_data_mt`

Reply:
After one of the recent patches it does (but it aligns to `chunk_size` not to `UCT_IB_MD_MAX_MR_SIZE`):

https://github.com/openucx/ucx/blob/master/src/uct/ib/mlx5/dv/ib_mlx5dv_md.c#L162

Reply:
Recent update should be good for both cases